### PR TITLE
Fix loading of parameters from `.env.*` files

### PR DESCRIPTION
**Pull request by smotesko**
The documentation [states](http://codeception.com/docs/06-ModulesAndHelpers#Dynamic-Configuration-With-Params) that we can load params from files named `.env` or `.env.something` but the latter doesn't work because the regex only matches filenames ending with `.env` or `.ini`

### CODE REVIEW

File: src/Codeception/Configuration.php [hunk 1]
```diff
@@ -646,7 +646,7 @@ private static function prepareParams($settings)
             }
 
             // .env and ini files
-            if (preg_match('~(\.ini|\.env)$~', $paramStorage)) {
+            if (preg_match('~(\.ini$|\.env)~', $paramStorage)) {
```
Comment (stipsan) [inline src/Codeception/Configuration.php]: why limit .ini files?

Comment (smotesko) [inline src/Codeception/Configuration.php]: @stipsan I see two reasons to "limit" .ini files:
1. They were "limited" before, so I don't change anything in that regard.
2. I've never seen a config file named `.ini.something`.

examples of file names that would match:
```
prod.ini
testing.ini
.env
.env.testing
```

Comment (stipsan) [inline src/Codeception/Configuration.php]: ok that makes sense regarding `.ini`.
I'm no regex wizard but I noticed a problem with your updated pattern though: https://regex101.com/r/hO5rU9/2

It's now matching stuff like `.envy` or anything with `.env` in it.

Comment (smotesko) [inline src/Codeception/Configuration.php]: @stipsan Yes, I'm aware of it. Don't think if would be a problem because that's the last condition in the chain of possible options (after the `.yml`). See, it doesn't match any files in the project root. It matches files that are explicitly specified in `codeception.yml`, so if user want to specify file named `.envy` there's no reason to deny it in this case, I think.

Comment (stipsan) [inline src/Codeception/Configuration.php]: I see, that's good. I think that's a developer friendly approach 👍

I don't know if this functionality is covered in the testing suite already, as I've only recently started to participate here, but I think adding a test that catch the edge case you're fixing in this PR would prevent future regressions.

Basically just a test that makes sure `.env.something` is loaded correctly :)

Comment (smotesko) [inline src/Codeception/Configuration.php]: @stipsan 👌 I've looked into covering this with test and I see it requires mocking the PHP functions. It can be done, but needs some thought to not make a mess. I'll have to do it later.

Comment (stipsan) [inline src/Codeception/Configuration.php]: that sounds complicated indeed.
Perhaps @DavertMik or @Naktibalda know more about this?

Comment (DavertMik) [inline src/Codeception/Configuration.php]: Thanks @stipsan for noticing it. 

I think next pattern should be used:

```
(\.ini$|\.env(\.|$))
```

**Comment by DavertMik** [comment]
Please update the regex

File: src/Codeception/Configuration.php (hunk 1 above)
Comment (smotesko) [inline src/Codeception/Configuration.php]: @DavertMik looks great, thanks. I've changed it.

**Comment by smotesko** [comment]
@DavertMik changed.

**Comment by DavertMik** [comment]
Thanks!
